source/code/plugins/VMInsightsDataCollector.rb

```diff
@@ -54,23 +54,28 @@ def get_available_memory_kb
             return available, total
         end
 
-        # returns: cummulative uptime, cummulative idle time
+        # returns: cummulative total time, cummulative idle time
+
+        # /proc/stat contains system statistics since last restart
+        # first line contains aggregate across all CPUs
+        # format:
+        #
+        # cpu user nice system idle iowait irq softirq steal guest guest_nice
+        # eg: cpu  2904083 315778 1613190 140077550 216726 0 88355 0 0 0
+        # https://www.kernel.org/doc/html/latest/filesystems/proc.html#miscellaneous-kernel-statistics-in-proc-stat
         def get_cpu_idle
-            uptime = nil
+            total_time = nil
             idle = nil
-            File.open(File.join(@root, "proc", "uptime"), "rb") { |f|
+            File.open(File.join(@root, "proc", "stat"), "rb") { |f|
                 line = f.gets
-                next if line.nil?
-                line.scanf(" %f %f ") { |u, i|
-                    uptime = u
-                    idle = i
+                raise Unavailable, "/proc/stat empty" if line.nil?
+                time_entries = line.split(" ")
+                # cpu user nice system idle - remaining entries depend on kernel version
+                raise Unavailable, "/proc/stat insufficient entries" if time_entries.length < 5
+                time_entries = time_entries.slice(1, time_entries.length) # skip the first entry in row: "cpu"
+                idle, total_time = time_entries[3].to_i, time_entries.map(&:to_i).sum
```

Comment:
This implementation does not match [get_cpu_time](https://msazure.visualstudio.com/InfrastructureInsights/_git/DependencyAgent?path=/collector/service/platform/linux/machine-metrics.cc&version=GBmaster&line=104&lineEnd=105&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contentsl)